pkg/agent/agent.go

```diff
@@ -206,7 +165,54 @@ func (k *Klocksmith) Run() error {
 	k.lc.Reboot(false)
 
 	// cross fingers
-	time.Sleep(24 * 7 * time.Hour)
+	sleepOrDone(24*7*time.Hour, stop)
+	return nil
```

Comment:
It doesn't seem like this function will ever actually return. `close(stop)` is done by the main function, but only once this function returns, which doesn't happen until the `stop` channel is closed. That doesn't seem like the intended behavior.